android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -916,11 +916,11 @@ protected void handleFocus()
 	}
 
 	@Kroll.method
-	public TiBlob toImage()
+	public TiBlob toImage(@Kroll.argument(optional=true) KrollFunction callback)
 	{
-		if (TiApplication.isUIThread()) {
-			return handleToImage();
-
+		if (callback != null) {
```

Comment:
The logic of this if statement is flawed b/c it doesn't take into account TiApplication.isUIThread() clause. Also, you should return the TiBlob, not null even when there's a callback present.